### ISSUE TITLE
Disconnect moves to singleplayer, then fails assert

### ISSUE THREAD

**Issue by SimonN** [SimonN/LixD#220]
Lix 0.8.4.

Play multiplayer. Produce severe lag that cuts the connection from the server. Get a crash in the game:

```
 15949.72 src/net/client.d:150:
 15949.72 call this function only when you're connected
 15949.72 ??:? _d_assert_msg [0xcc5fc7da]
??:? const(pure @property const(net.structs.Profile) function()) net.client.NetClient.ourProfile [0xcc5d6cf8]
??:? pure game.core.view.View game.core.view.createView(const(int), const(net.iclient.INetClient)) [0xcc55616b]
??:? const(@property game.core.view.View function()) game.core.game.Game.view [0xcc5541bb]
??:? void game.core.calc.implGameCalc(game.core.game.Game) [0xcc5515c3]
??:? void game.core.game.Game.calc() [0xcc553dcc]
??:? void basics.mainloop.MainLoop.calc() [0xcc44e19f]
??:? void basics.mainloop.MainLoop.mainLoop() [0xcc44d827]
??:? int main.main(immutable(char)[][]).__lambda2() [0xcc5c4e4c]
??:? extern (C) int allegro5.system.al_run_allegro(scope int delegate()).main_runner(int, char**) [0xcc5f8602]
??:? int allegro5.system.al_run_allegro(scope int delegate()) [0xcc5f85dd]
??:? _Dmain [0xcc5c4de3]
```

**Comment by SimonN** [comment]
I don't know what happens in release mode. I feel that #220 is responsible for [this crash by Forestidia](https://www.lemmingsforums.net/index.php?topic=3710.msg69317#msg69317).

Let's focus on #220! I've put it off for long enough. Detect the connection drop and report it on the screen, cancelling the game.

**Comment by SimonN** [comment]
Will be fixed in 0.9.11.

All connection losses drop you back before the lobby, and you get to click on Connect again. The game won't offer a dialog box to save the replay.

When your option to save all multiplayer replays is enabled (on by default), your replay is saved into `replays/network` like a finished game.
